operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -307,6 +347,72 @@ func (r *TenantReconciler) checkValidWorkspaces(ctx context.Context, tn *crownla
 	return tenantExistingWorkspaces, workspaces, err
 }
 
+// deleteClusterNamespace deletes the namespace for the tenant, if it fails then it returns an error
+func (r *TenantReconciler) deleteClusterNamespace(ctx context.Context, tn *crownlabsv1alpha2.Tenant, nsName string) (nsOk bool, err error) {
+	ns := v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nsName}}
+
+	nsErr := utils.EnforceObjectAbsence(ctx, r.Client, &ns, "personal namespace");
+    
+	if nsErr != nil {
+		klog.Errorf("Error when deleting namespace of tenant %s -> %s", tn.Name, nsErr)
+		return false, nsErr
+	}
+
+	r.updateTnNamespace(&ns, tn.Name)
```

Comment:
This operation shall not be performed, since the namespace has just been deleted.

Reply:
Resolved in [0d41a92](https://github.com/netgroup-polito/CrownLabs/pull/821/commits/0d41a922884c6a5c86189830aec583fa59690a19)